test/decrypter.test.js

```diff
@@ -47,47 +61,103 @@ QUnit.test('decrypts multiple AES-128 blocks with CBC', function() {
+
+  assert.expect(1);
+
+  decrypt(encrypted, key, initVector, function(err, result) {
+    if (err) {
+      QUnit.notOk(true, 'Non-null error.');
+    }
+    QUnit.deepEqual(stringFromBytes(result),
+                    '0123456789abcdef01234',
+                    'decrypted multiple blocks'
+                   );
+    done();
+  });
 });
 
 QUnit.test(
-'verify that the deepcopy works by doing two decrypts in the same test',
-function() {
-  let key = new Uint32Array([0, 0, 0, 0]);
-  let initVector = key;
-  // the string "howdy folks" encrypted
-  let pkcs7Block = new Uint8Array([
-    0xce, 0x90, 0x97, 0xd0,
-    0x08, 0x46, 0x4d, 0x18,
-    0x4f, 0xae, 0x01, 0x1c,
-    0x82, 0xa8, 0xf0, 0x67
-  ]);
-
-  QUnit.deepEqual('howdy folks',
-                  stringFromBytes(unpad(decrypt(pkcs7Block, key, initVector))),
-                  'decrypted with a byte array key'
-  );
-
-// the string "0123456789abcdef01234" encrypted
-  let cbcBlocks = new Uint8Array([
-    0x14, 0xf5, 0xfe, 0x74,
-    0x69, 0x66, 0xf2, 0x92,
-    0x65, 0x1c, 0x22, 0x88,
-    0xbb, 0xff, 0x46, 0x09,
-
-    0x0b, 0xde, 0x5e, 0x71,
-    0x77, 0x87, 0xeb, 0x84,
-    0xa9, 0x54, 0xc2, 0x45,
-    0xe9, 0x4e, 0x29, 0xb3
-  ]);
+  'verify that the deepcopy works by doing two decrypts in the same test',
+  function(assert) {
+    assert.expect(2);
+
+    let done = assert.async(2);
+    let key = new Uint8Array(16);
+    let initVector = key;
+
+    // the string "howdy folks" encrypted
+    let pkcs7Block = new Uint8Array([
+      0xce, 0x90, 0x97, 0xd0,
+      0x08, 0x46, 0x4d, 0x18,
+      0x4f, 0xae, 0x01, 0x1c,
+      0x82, 0xa8, 0xf0, 0x67
+    ]);
+
+    decrypt(pkcs7Block, key, initVector, function(err, result) {
+      if (err) {
+        QUnit.notOk(true, 'Non-null error.');
+      }
+      QUnit.deepEqual(stringFromBytes(result),
+                      'howdy folks',
+                      'decrypted with a byte array key'
+                     );
+      done();
+    });
+
+    // the string "0123456789abcdef01234" encrypted
+    let cbcBlocks = new Uint8Array([
+      0x14, 0xf5, 0xfe, 0x74,
+      0x69, 0x66, 0xf2, 0x92,
+      0x65, 0x1c, 0x22, 0x88,
+      0xbb, 0xff, 0x46, 0x09,
+
+      0x0b, 0xde, 0x5e, 0x71,
+      0x77, 0x87, 0xeb, 0x84,
+      0xa9, 0x54, 0xc2, 0x45,
+      0xe9, 0x4e, 0x29, 0xb3
+    ]);
+
+    decrypt(cbcBlocks, key, initVector, function(err, result) {
+      if (err) {
+        QUnit.notOk(true, 'Non-null error.');
+      }
+      QUnit.deepEqual(stringFromBytes(result),
+                      '0123456789abcdef01234',
+                      'decrypted multiple blocks'
+                     );
+      done();
+    });
+  }
+);
 
-  QUnit.deepEqual('0123456789abcdef01234',
-                  stringFromBytes(unpad(decrypt(cbcBlocks, key, initVector))),
-                  'decrypted multiple blocks');
+QUnit.test('asynchronously decrypts a 4-word block', function(assert) {
+  assert.expect(1);
 
+  let done = assert.async();
```

Comment:
For simple timeout stuff, our standard is to use sinon's clock.tick() instead of async tests.